CSharpFunctionalExtensions/Result/Extensions/TapAsyncLeft.cs

```diff
@@ -49,5 +49,24 @@ public static async Task<Result<T>> Tap<T>(this Task<Result<T>> resultTask, Acti
             Result<T, E> result = await resultTask.DefaultAwait();
             return result.Tap(action);
         }
+
+
+        /// <summary>
+        ///     Executes the given action if the calling result is a success. Returns the calling result on success.
+        /// </summary>
+        public static async Task<Result<T>> Tap<T, K>(this Task<Result<T>> resultTask, Func<T, Result<K>> func)
+        {
+            Result<T> result = await resultTask.DefaultAwait();
+            return result.Bind(func).Map(_ => result.Value);
```

Comment:
Can we not just `return result.Tap(action);` like the others here?

Reply:
I thought that `Tap` signatures with `Result`-returning `Func`s were the only signatures that would behave like `TapAndCheck`, where the `Action` signatures would continue to ignore the result altogether. Wouldn't changing this to `return result.Tap(action)` cause us to call the wrong overload?

Reply:
No, the `func` parameter is still a `Func` and this just passes it to the non-async `Tap()` overload.

Reply:
I should have said `return result.Tap(func)` of course - sorry, missed that typo.

Reply:
Got it. Thanks for the clarification.